src/search/Search.tsx

```diff
@@ -1,7 +1,57 @@
+import { useCallback, useEffect, useRef } from 'react';
+
+import { useSearchParams } from 'react-router-dom';
+
+import Spacing from '@/components/Spacing/Spacing';
+
+import { ResultListItem } from './components/ResultListItem/ResultListItem';
+import { SearchBar } from './components/SearchBar/SearchBar';
+import { useGetSearch } from './hooks/useGetSearch';
+import { NoResult } from './pages/NoResult/NoResult';
+
 export const Search = () => {
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get('query');
+
+  const { data, isLoading, fetchNextPage, hasNextPage } = useGetSearch({
+    query: query as string,
+  });
```

Comment:
검색어를 입력하지 않은 경우 `query`가 null일 수 있어서 라인 17에서 string으로 단언하신 거 같은데 맞나요?

혹시 `||` 연산자 써서 명시적으로 null 체크하는 방법은 어떨까요
query가 null인 경우가 발생하지 않는 게 보장된 상황이면 단언문 써도 될 거 같은데 그게 아니니까...!!

```suggestion
  const query = searchParams.get('query') || '';

  const { data, isLoading, fetchNextPage, hasNextPage } = useGetSearch({
    query: query,
  });
```

Reply:
왕 보리 ,, 좋습니다 ! 이런 방식이 있는지 몰랐네요
지식 +1.. 감사합니다 🙇🏻‍♀️